### PR TITLE
List usecases on the index page

### PR DESCRIPTION
**Pull request by geek-yang**
- create components for index page
- create a md file to host the entries of usecases
- load usecases data from md file and export data to the usecase card component
- link each usecase card to the actual usecase page

### CODE REVIEW

File: components/UseCaseCard.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable */
+<template>
+  <div class="flex p-4">
+    <!-- Usecase cards -->
+    <div class="flex no-wrap text-left">
+      <h4>
+        <NuxtLink :to="card.route">
```
Comment (Peter9192) [inline components/UseCaseCard.vue]: I suppose you could also wrap the entire card in a nuxt-link, rather than the title and image separately.

File: components/Banner.vue [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<template>
```
Comment (Peter9192) [inline components/Banner.vue:1]: Good idea to make a banner component!

File: content/usecaseItems.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (Peter9192) [inline content/usecaseItems.md:1]: It feels a bit like we're repeating the yaml headers of the individual pages here. It would be nicer if we could somehow fetch them automatically. 

On the other hand.. this file currently indexes (all the pages of) a single use case. However, as we discussed, we also need an overview of all use cases. Maybe we can repurpose this file for that.

File: pages/index.vue [hunk 4]
```diff
@@ -1,9 +1,53 @@
...
+    </div>
+
+    <div class="flex overflow-auto">
+      <!-- usecases -->
+      <div class="m-2 bg-white rounded">
+        <!-- category 1 -->
+        <h3 class="text-2xl">
+          Category 1
+        </h3>
+        <div class="flex text-left">
```
Comment (Peter9192) [inline pages/index.vue]: This now adds one usecasecard *for each page* of a use case. I think we are aiming for one usecasecard per usecase (which could potentially include a carousel/image slider so as to incorporate all pages in that single card).

File: pages/index.vue [hunk 5]
```diff
@@ -1,9 +1,53 @@
...
+            v-for="usecase in usecases.items"
+            :key="usecase.id"
+            :card="usecase"
+          />
+        </div>
+        <!-- category 2 -->
+        <h3 class="text-2xl">
+          Category 2
+        </h3>
+      </div>
```
Comment (Peter9192) [inline pages/index.vue]: Maybe we can skip the catogories for now and just add them later on once we really need them. But good to see that it's quite straightforward

File: pages/index.vue [hunk 6]
```diff
@@ -1,9 +1,53 @@
...
+
+      <!-- Tags -->
+      <div class="m-2 bg-white rounded">
+        <h3>
+          Tags
```
Comment (Peter9192) [inline pages/index.vue:26]: This now looks a bit strange, because we don't have tags (yet). Maybe we can add some dummy tags, like "CPM data", "spatial merging", and "constraints"?

Maybe add some styling as well, similar to e.g. https://cdn.business2community.com/wp-content/uploads/2014/01/tags.jpg

File: pages/index.vue [hunk 7]
```diff
@@ -1,9 +1,53 @@
...
+    </div>
+  </div>
 </template>
 
 <script>
-export default {}
+import Banner from '~/components/Banner.vue'
+import UseCaseCard from '~/components/UseCaseCard.vue'
+export default {
+  components: { Banner, UseCaseCard },
```
Comment (Peter9192) [inline pages/index.vue]: I think we don't need these imports here. See https://dev.to/ryosh/use-nuxt-components-to-automatically-import-components-51cf and https://github.com/nuxt/components

**Review by Peter9192 — COMMENTED** [review]
Hi @geek-yang great work on this feature already! I think we need to agree on a process of what to do before merging new features etc.. Let's talk about that soon. Perhaps you could continue to work on this branch until then? 

For now, I've added some comments, some of which we've already discussed. Hope this helps!

File: components/UseCaseCard.vue (hunk 1 above)
Comment (geek-yang) [reply components/UseCaseCard.vue]: Good point! It is resolved.

File: components/Banner.vue (hunk 2 above)
Comment (geek-yang) [reply components/Banner.vue:1]: Thanks! We can reuse it later.

File: content/usecaseItems.md (hunk 3 above)
Comment (geek-yang) [reply content/usecaseItems.md:1]: After some googling, I find the easiest and most clean way to fetch data and feed them to the component would be using a md file including a list of items, rather than retrieving the yaml file and looping through all the information from individual md files (or maybe there is a nice way to do this but I did not find). We can keep it for now as it is always nice to have an overview of all use cases, and change it when it is necessary.

File: pages/index.vue (hunk 5 above)
Comment (geek-yang) [reply pages/index.vue]: Sounds good! I remove the other categories.

File: pages/index.vue (hunk 4 above)
Comment (geek-yang) [reply pages/index.vue]: Good idea! We will test/implement the carousel display in another PR.

File: pages/index.vue (hunk 7 above)
Comment (geek-yang) [reply pages/index.vue]: Thanks for mentioning this! I remove these lines and now it looks much clean!

File: pages/index.vue (hunk 6 above)
Comment (geek-yang) [reply pages/index.vue:26]: Agree. But I think it is better that we leave it there as we will need it later. Just take it into consideration when doing styling. I will add some dummy tags in a separate PR.

**Comment by geek-yang** [comment]
Resolve all the comments and polish the code a bit. Ready for review for another round now!

File: content/usecaseItems.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+case_0_id: 0
+case_0_panel:
+  # - title: 
+  #  tooltip: Home
+  #  route: /usecase_0  
+  - title: Conceptual overview of the use case
```
Comment (Peter9192) [inline content/usecaseItems.md:7]: I think this would have to be a name that's descriptive for the entire usecase, not just the first panel. Maybe "Example use case" is a better fit?

File: package.json [hunk 9]
```diff
@@ -13,7 +13,8 @@
   "dependencies": {
     "@nuxt/content": "^1.14.0",
     "core-js": "^3.15.1",
-    "nuxt": "^2.15.7"
+    "nuxt": "^2.15.7",
+    "vueperslides": "^3.3.1"
```
Comment (Peter9192) [inline package.json:17]: I see how this ended up here, but actually it's not really part of this PR. I'm okay with leaving it in for now.

**Review by Peter9192 — COMMENTED** [review]
Very nice! Just a small edit and it's good to go

File: package.json (hunk 9 above)
Comment (geek-yang) [reply package.json:17]: I think this is when I tested the carousel, the package is added automatically. Let's leave it for now.

File: content/usecaseItems.md (hunk 8 above)
Comment (geek-yang) [reply content/usecaseItems.md:7]: Thanks for the comment. I rename it as "example_usecase".

**Review by Peter9192 — APPROVED** [review]
I'll merge it now, let's see the automatic deployment in action ;-)
